Sources/Datadog/DatadogConfiguration.swift

```diff
@@ -505,6 +507,17 @@ extension Datadog {
                 return self
             }
 
+            /// Sets the Session start callback.
+            ///
+            /// The callback takes 2 arguments: the newly started Session ID and a boolean indicating whether or not the session is discarded by the sampling rate
+            /// (when `true` it means no event in this session will be kept).
+            ///
+            /// - Parameter handler: the callback handler to notify whenever a new Session starts.
+            public func onSessionStart(_ handler: @escaping (String, Bool) -> Void) -> Builder {
```

Comment:
- [ ] About the naming - as this is a RUM feature, I think it's worth expressing it from the name as we do for other RUM options. How about `onRUMSessionStart`?
- [ ] About testing - it also requires updating tests in [`DatadogConfigurationBuilderTests`](https://github.com/DataDog/dd-sdk-ios/blob/f571f041edb2b1097b6c4fd28bdcdfdf2408ca66/Tests/DatadogTests/Datadog/DatadogConfigurationBuilderTests.swift#L15) (we cover all public options in there)